libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -738,7 +743,12 @@ - (void)finalizeAuthCompletion
 {
     // Apply the credentials that will ensure there is a user and that this
     // current user as the proper credentials.
-    SFUserAccount *user = [[SFUserAccountManager sharedInstance] applyCredentials:self.coordinator.credentials];
+    SFUserAccountChange change = [self userAccountChangeFromDict:self.coordinator.credentialsChangeSet];
```

Comment:
This seems like an area of responsibility for `SFUserAccountManager`.  Why not make this assessment there?